src/services/storage.js

```diff
@@ -0,0 +1,17 @@
+//this service is made to handle all localstorage variables
+const NEWSLETTERKEY = "newsletter";
+let newsLetterValue = localStorage.getItem(NEWSLETTERKEY);
+
+export const getNewsletterSubcription = () => {
+  return newsLetterValue;
+};
+
+export const subscribeNewsletter = () => {
+  localStorage.setItem(NEWSLETTERKEY, true);
+  newsLetterValue = true;
+};
+
+export const unsubscribeNewsletter = () => {
+  localStorage.removeItem(NEWSLETTERKEY);
+  newsLetterValue = false;
+};
```

Comment:
Using local storage is a much better idea than using cookies 👍 .

The "cookie law" [applies to local storage as well](https://www.cookielaw.org/the-cookie-law/) so they'll need to be a [snackbar](https://material-ui.com/components/snackbars/#snackbar) notification displayed to the user.

Using a service is a good plan. Here's a suggested implementation (self documenting, no mutation, descriptive variable names, extendable).

```suggestion
export const items = {
  isSubscribedToNewsletter: 'isSubscribedToNewsletter',
};

export default {
  getItem: (item) => localStorage.getItem(item),
  setItem: ({ item, value }) => localStorage.setItem(item, value),
};
```

_Usage_
```
import storage, { items } from 'services/storage'

// Signs up
storage.setItem({ item: items.isSubscribedToNewsletter, value: true })

// App mounts
const isSubscribedToNewsletter = storage.getItem(items.isSubscribedToNewsletter)
...
```

It would also be good to test this service.

- [ ] Create snackbar
- [ ] Generic implementation
- [ ] Add tests